magento/ece-tools
# MCLOUD-5941: Applying patches is happening too late

### Description
https://jira.corp.magento.com/browse/MCLOUD-5941
The apply-patches step should have the highest priority during deployment.

>**Applying patches is happening too late**
Since di:compile and module:refresh may fail due to Magento issues, patches must be applied on earlier stage
AC
Move patch applying step before di:compile and module:refresh

### Fixed Issues (if relevant)
https://jira.corp.magento.com/browse/MCLOUD-5941

### Manual testing scenarios
1. Launch Deployment (with build phase)
2. After successful deployment, open the log file and make sure that the recording of the apply patches was immediately after the recording of the start of the scenario 'generate':

```
[datetime] INFO: Starting scenario(s): scenario/build/generate.xml (magento/ece-tools version: 2002.1.0, magento/magento2-base version: 2.3.3-p1)  
[datetime] INFO: Verbosity level is not set  
[datetime] NOTICE: Applying patches  
[datetime] NOTICE: End of applying patches
[datetime] INFO: Set Magento application mode to 'production'  
[datetime] NOTICE: Validating configuration
```

### Contribution checklist
